es/es-server/src/main/java/org/elasticsearch/Version.java

```diff
@@ -44,26 +44,38 @@
-    public static final Version V_5_4_3 = new Version(V_5_4_3_ID, org.apache.lucene.util.Version.LUCENE_6_5_1);
-    public static final int V_5_5_2_ID = 5050299;
-    public static final Version V_5_5_2 = new Version(V_5_5_2_ID, org.apache.lucene.util.Version.LUCENE_6_6_0);
-    public static final int V_5_6_3_ID = 5060399;
-    public static final Version V_5_6_3 = new Version(V_5_6_3_ID, org.apache.lucene.util.Version.LUCENE_6_6_1);
-    public static final int V_5_6_8_ID = 5060899;
-    public static final Version V_5_6_8 = new Version(V_5_6_8_ID, org.apache.lucene.util.Version.LUCENE_6_6_1);
-    public static final int V_6_1_4_ID = 6010499;
-    public static final Version V_6_1_4 = new Version(V_6_1_4_ID, org.apache.lucene.util.Version.LUCENE_7_1_0);
-    public static final int V_6_5_1_ID = 6050199;
-    public static final Version V_6_5_1 = new Version(V_6_5_1_ID, org.apache.lucene.util.Version.LUCENE_7_5_0);
-
-    public static final Version CURRENT = V_6_5_1;
+    public static final Version V_EMPTY = new Version(V_EMPTY_ID, 0, org.apache.lucene.util.Version.LATEST);
+    public static final int ES_V_5_0_2_ID = 5000299;
+    public static final Version ES_V_5_0_2 = new Version(ES_V_5_0_2_ID, 2_00_04_99, org.apache.lucene.util.Version.LUCENE_6_2_1);
+    public static final int ES_V_5_2_2_ID = 5020299;
+    public static final Version ES_V_5_2_2 = new Version(ES_V_5_2_2_ID, 2_01_00_99, org.apache.lucene.util.Version.LUCENE_6_4_1);
+    public static final int ES_V_5_4_3_ID = 5040399;
+    public static final Version ES_V_5_4_3 = new Version(ES_V_5_4_3_ID, 2_01_01_99, org.apache.lucene.util.Version.LUCENE_6_5_1);
+    public static final int ES_V_5_5_2_ID = 5050299;
+    public static final Version ES_V_5_5_2 = new Version(ES_V_5_5_2_ID, 2_02_00_99, org.apache.lucene.util.Version.LUCENE_6_6_0);
+    public static final int ES_V_5_6_3_ID = 5060399;
+    public static final Version ES_V_5_6_3 = new Version(ES_V_5_6_3_ID, 2_03_00_99, org.apache.lucene.util.Version.LUCENE_6_6_1);
+    public static final int ES_V_5_6_8_ID = 5060899;
+    public static final Version ES_V_5_6_8 = new Version(ES_V_5_6_8_ID, 2_03_04_99, org.apache.lucene.util.Version.LUCENE_6_6_1);
+    public static final int ES_V_6_1_4_ID = 6010499;
+    public static final Version ES_V_6_1_4 = new Version(ES_V_6_1_4_ID, 3_00_01_99, org.apache.lucene.util.Version.LUCENE_7_1_0);
+    public static final int ES_V_6_5_1_ID = 6050199;
+    public static final Version ES_V_6_5_1 = new Version(ES_V_6_5_1_ID, 3_02_00_99, org.apache.lucene.util.Version.LUCENE_7_5_0);
+
+    /**
+     * Before CrateDB 4.0 we've had ES versions (internalId) and CrateDB (externalId) versions.
+     * The internalId is stored in indices, so we keep using it for compatibility.
+     *
+     * All internalIds are mapped to the first CrateDB version which shipped with the given ES version.
+     * Starting from internalId 7.0.0, we've a static offset as we are always increasing both versions in sync.
```

Comment:
this is not entirely clear "we've a static offset as we are always increasing both versions in sync". 

Does this mean we'll never do a crate release without increasing the ES version (ie. upgrading an ES piece?) Or did you mean we'll always keep the offset and the crate (external) version in sync?

Reply:
@andreidan pushed a fixup that should clarify it